Assets/Scripts/Audio/AudioSpectrumDrawer.cs

```diff
@@ -0,0 +1,94 @@
+using UnityEngine;
+
+using ProefExamen.Audio.WaveFormDrawer;
+
+namespace ProefExamen.Audio.SpectrumDrawer
+{
+    /// <summary>
+    /// Class which is used for drawing the audio spectrum of a song.
+    /// </summary>
+    public class AudioSpectrumDrawer : MonoBehaviour
+    {
+        [SerializeField]
+        private int _audioSamples = 512;
+
+        [SerializeField]
+        private float _lerpAmount = 20;
+
+        [SerializeField]
+        private float _maxScale = 1200;
+
+        [SerializeField]
+        private Vector3 _offsetPosition = Vector3.zero;
+
+        [Space]
+        [SerializeField]
+        private GameObject _imagePrefab;
+
+        private GameObject[] _visualizers;
+        private AudioWaveformDrawer _waveformDrawer;
+
+        private float[] _samples;
+
+        private void Awake()
+        {
+            _samples = new float[_audioSamples];
+            _waveformDrawer = FindObjectOfType<AudioWaveformDrawer>();
+        }
+
+        /// <summary>
+        /// Method which analyses the specified audio clip's samples.
+        /// </summary>
+        /// <param name="clip">Clip used for analysing the audio spectrum data.</param>
+        public void VisualizeSongSpectrum(AudioClip clip)
+        {
+            _visualizers = null;
+            foreach (Transform child in transform)
+                Destroy(child.gameObject);
+
+            _waveformDrawer.InitializeDrawer(clip);
+            _visualizers = new GameObject[_samples.Length];
+
+            CreateBandVisualizers();
+        }
+
+        /// <summary>
+        /// Method used for creating band visualizers.
+        /// </summary>
+        private void CreateBandVisualizers()
+        {
+            for (int i = 0; i < _visualizers.Length; i++)
+            {
+                GameObject newBandVisualizer = Instantiate(_imagePrefab, transform);
+                newBandVisualizer.transform.position = transform.position;
+                newBandVisualizer.transform.parent = transform;
+
+                newBandVisualizer.name = "BandVisualizer " + i;
+                newBandVisualizer.transform.position = new Vector3(transform.position.x + (i * transform.localScale.x), transform.position.y, 0);
```

Comment:
This line exceeds the 120 character limit.
```suggestion
                newBandVisualizer.transform.position = new Vector3(
                    transform.position.x + (i * transform.localScale.x),
                    transform.position.y, 0
                );
```